### ISSUE TITLE
Glare can't compile with DirectX 12 SM6

### ISSUE THREAD

**Issue by MakotoIchinose** [EscapeEntertainment/PrettyPostProcess#1]
Crash log:
```
[REDACTED]/Engine/Plugins/Runtime/PrettyPostProcess/Shaders/Glare.usf(): Shader FGlareGS, Permutation 0, VF None:
	Internal Compiler error: llvm::cast<X>() argument of incompatible type!

[REDACTED]/Engine/Plugins/Runtime/PrettyPostProcess/Shaders/Glare.usf(): Shader FGlareGS, Permutation 0, VF None:
	D3DCompileToDxil failed. Error code: Unknown error (0x80AA001D).
```

**Comment by MakotoIchinose** [comment]
On a weird turn of event, the Glare shaders can compile in SM5 even with RT on, but can't compile in SM6 even with RT off.

**Comment by AsahiChan-Game** [comment]
I have met the same problem recently. Is there any solution?

**Comment by MakotoIchinose** [comment]
> I have met the same problem recently. Is there any solution?

Not yet. The error was too cryptic, and I haven't found way to debug the DXIL compiler.

**Comment by MakotoIchinose** [comment]
Fixed for 5.3 per e954008
Finally the error become decipherable
